libc/src/__support/complex_type.h

```diff
@@ -16,5 +16,12 @@ template <typename T> struct Complex {
   T real;
   T imag;
 };
+
+template <typename T, typename U> T conjugate(T c) {
```

Comment:
I am not sure I see the benefit of this indirection here. We would then also have to add header guards for cfloat16 and cfloat128 in complex_types.h which I think is unnecessary. I think we should leave the functions in complex_types.h as generic as possible.

Reply:
True, but I think that it would be trivial for anyone using this function to deduce the order of complex and real type while using the templated form. Regarding cluttering, I don't think 2 parameters is too much cluttering. And my point regarding extra guards, the extra guards in complex_types.h would be redundnant because we would have only called the template form with cfloat16 if it existed.

Reply:
So the complexity of 2 template parameters will expanded down the line when more functions are implemented, and it will hinder template argument deduction if both types are not presented in the arguments.  An apt example would be imagine every template function implementation for `float` will also have to pass the corresponding `FPBits` type to the template parameter, instead of using `FPBits<float>` inside the implementation.  If you don't want to put those interchanging template utilities in `complex_type.h`, feel free to add another header in `__support/CPP/type_traits/` and include that here.

Reply:
oh okay, I see your point. Added them in complex_types.h